src/WooCommerce.php

```diff
@@ -64,6 +66,15 @@ public static function getProductDetails($product_id = 0, $primary_category = TR
       $category = static::getProductCategoriesParentsList($product_id);
     }
 
+    if (empty($category)) {
+      if ($primary_category) {
+        $category = static::getProductCategoryParents(static::getProductPrimaryCategoryId($parent_id), '/');
```

Comment:
Not every product has a parent
```suggestion
        $category = static::getProductCategoryParents(static::getProductPrimaryCategoryId($parent_id ?: $product_id), '/');
```

Reply:
You can embed this into the previous lines instead of repeating them. Saves performance (less queries).